### ISSUE TITLE
[Suggestion] blacklist for individual books as well

### ISSUE THREAD

**Issue by jeremiahwinsley** [EccentricVamp/EccentricTome#11]
**Is your feature request related to a problem? Please describe.**

Some mods like Apotheosis and Dark Utilities add single use books, like `Tome of Helmets` or `Galactic Tome`.  These are deleted when used. So if someone adds these to their tome and uses them, it would delete the whole thing.

Currently, you can blacklist a whole mod, but there is no way to only allow specific items from a mod.

**Describe the solution you'd like**
One of:
- blacklist a mod, but whitelist one or more items from it.
- blacklist a set of items from a mod
- blacklist single use items automatically (but I don't think there's an easy way to check for this)

**Comment by EccentricVamp** [comment]
> So if someone adds these to their tome and uses them, it would delete the whole thing
This is a good point. I'll add a warning to the readme and Curse Forge project description.

The simplest solution for me to implement would be blacklisting specific items by their resource locations (ids), similar to how the existing item whitelist works.

I might circle back around later and implement more complex black/white listing like the first solution you described.

**Comment by EccentricVamp** [comment]
@jeremiahwinsley could you provide a list of resource locations (like `<mod_id>:<item_id>` ) for the _Tome of Helmets_ and _Galactic Tome_?

I could hunt them down on my own, but neither of those mods are part of my current pack.

Alternatively, you're welcome to open up a pull request and add them yourself. [Here's an example](https://github.com/EccentricVamp/EccentricTome/pull/9) of how the pull request might look.

**Comment by jeremiahwinsley** [comment]
Here's the ones I'm aware of from ATM7:
```
apotheosis:boots_tome
apotheosis:bow_tome
apotheosis:chestplate_tome
apotheosis:fishing_tome
apotheosis:helmet_tome
apotheosis:leggings_tome
apotheosis:other_tome
apotheosis:pickaxe_tome
apotheosis:scrap_tome
apotheosis:weapon_tome
darkutils:tome_enchanting
darkutils:tome_illager
darkutils:tome_pigpen
darkutils:tome_runelic
darkutils:tome_sga
darkutils:tome_shadows
minecolonies:ancienttome
occultism:book_of_binding_afrit
occultism:book_of_binding_bound_afrit
occultism:book_of_binding_bound_djinni
occultism:book_of_binding_bound_foliot
occultism:book_of_binding_bound_marid
occultism:book_of_binding_djinni
occultism:book_of_binding_foliot
occultism:book_of_binding_marid
occultism:book_of_calling_djinni_manage_machine
occultism:book_of_calling_foliot_cleaner
occultism:book_of_calling_foliot_lumberjack
occultism:book_of_calling_foliot_transport_items
```

Possibly also `quark:ancient_tome` but I didn't test that.

**Comment by jeremiahwinsley** [comment]
FYI, I updated to 1.2.0 to test the blacklist, and:
- I was not able to add any books, regardless of blacklist status.
- if you place the Eccentric Tome and a book in the crafting grid (tested with Apotheosis Chronicle of Shadows), click on the (blank) output, and then drag the Eccentric Tome back to the inventory without moving the other book, it will delete the tome.
- `defineList` should be `defineListAllowEmpty` on the config, otherwise NightConfig parses it incorrectly. See relevant discussion here in the Forge discord: https://discord.com/channels/313125603924639766/867851603468615740/903378854670925905

**Comment by EccentricVamp** [comment]
Thanks for the list and the heads up! I've marked 1.2.0 as an alpha release until I can do more testing.

I'll try `defineListAllowEmpty` but it doesn't accept `List<String>` for its arguments so I'll have to figure out a way around that

**Comment by jeremiahwinsley** [comment]
Here's a reference from one of my mods if it's helpful:

https://github.com/jeremiahwinsley/pylons/blob/06035724d21cad81656d591ec5581d5ad5a8be87/src/main/java/net/permutated/pylons/ConfigManager.java#L86

**Comment by EccentricVamp** [comment]
That definitely helped. This feature has now been implemented in d65b300d5cc3f76c4bd6c85afa70efc391c81931 and 6ea4c36acbda2e1df588b61e141c2bc780683480 and published in version 1.2.1
